pydra/tasks/mriqc/workflows/anatomical/base.py

```diff
@@ -578,401 +578,375 @@ def headmsk_wf(name="HeadMaskWorkflow", omp_nthreads=1):
+#             function=gradient_threshold,
+#         ),
+#         name="GradientThreshold",
+#         num_threads=omp_nthreads,
+#     )
+#     if config.workflow.species != "human":
+#         gradient.inputs.sigma = 3.0
+#         thresh.inputs.aniso = True
+#         thresh.inputs.thresh = 4.0
+
+#     apply_mask = pe.Node(ApplyMask(), name="apply_mask")
+
+#     # fmt: off
+#     workflow.connect([
+#         (inputnode, enhance, [("in_file", "in_file"),
+#                               (("in_tpms", _select_wm), "wm_tpm")]),
+#         (inputnode, thresh, [("brainmask", "brainmask")]),
+#         (inputnode, gradient, [("brainmask", "brainmask")]),
+#         (inputnode, apply_mask, [("brainmask", "in_mask")]),
+#         (enhance, gradient, [("out_file", "in_file")]),
+#         (gradient, thresh, [("out_file", "in_file")]),
+#         (enhance, apply_mask, [("out_file", "in_file")]),
+#         (thresh, outputnode, [("out_file", "out_file")]),
+#         (apply_mask, outputnode, [("out_file", "out_denoised")]),
+#     ])
+#     # fmt: on
+
+#     return workflow
+## draft conversion up to here 19/13/2024
+
+
+## draft conversion of the following section to the end(updated 26/03/2024)
 
 def airmsk_wf(name="AirMaskWorkflow"):
-    """
-    Calculate air, artifacts and "hat" masks to evaluate noise in the background.
-
-    This workflow mostly addresses the implementation of Step 1 in [Mortamet2009]_.
-    This work proposes to look at the signal distribution in the background, where
-    no signals are expected, to evaluate the spread of the noise.
-    It is in the background where [Mortamet2009]_ proposed to also look at the presence
-    of ghosts and artifacts, where they are very easy to isolate.
-
-    However, [Mortamet2009]_ proposes not to look at the background around the face
-    because of the likely signal leakage through the phase-encoding axis sourcing from
-    eyeballs (and their motion).
-    To avoid that, [Mortamet2009]_ proposed atlas-based identification of two landmarks
-    (nasion and cerebellar projection on to the occipital bone).
-    MRIQC, for simplicity, used a such a mask created in MNI152NLin2009cAsym space and
-    projected it on to the individual.
-    Such a solution is inadequate because it doesn't drop full in-plane slices as there
-    will be a large rotation of the individual's tilt of the head with respect to the
-    template.
-    The new implementation (23.1.x series) follows [Mortamet2009]_ more closely,
-    projecting the two landmarks from the template space and leveraging
-    *NiTransforms* to do that.
-
-    .. workflow::
-
-        from mriqc.testing import mock_config
-        from mriqc.workflows.anatomical.base import airmsk_wf
-        with mock_config():
-            wf = airmsk_wf()
-
-    """
-    workflow = pe.Workflow(name=name)
+    workflow = pydra.Workflow(name=name)
 
-    inputnode = pe.Node(
-        niu.IdentityInterface(
+    inputnode = pydra.Node(
```

Comment:
The "inputnode" and "outputnode" tasks are no longer necessary in Pydra so can be commented out. The input fields (e.g. "in_file" and "in_mask" in this case) should be added to the "input_spec" kwarg of the `pydra.Workflow` init